server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookCallerImplTest.java

```diff
@@ -42,69 +45,128 @@
 
     RecordedRequest recordedRequest = server.takeRequest();
     assertThat(recordedRequest.getMethod()).isEqualTo("POST");
     assertThat(recordedRequest.getPath()).isEqualTo("/ping");
-    assertThat(recordedRequest.getBody().readUtf8()).isEqualTo(payload.getJson());
+    assertThat(recordedRequest.getBody().readUtf8()).isEqualTo(PAYLOAD.getJson());
     assertThat(recordedRequest.getHeader("User-Agent")).isEqualTo("SonarQube/6.2");
     assertThat(recordedRequest.getHeader("Content-Type")).isEqualTo("application/json; charset=utf-8");
-    assertThat(recordedRequest.getHeader("X-SonarQube-Project")).isEqualTo(payload.getProjectKey());
+    assertThat(recordedRequest.getHeader("X-SonarQube-Project")).isEqualTo(PAYLOAD.getProjectKey());
   }
 
   @Test
   public void silently_catch_error_when_external_server_does_not_answer() throws Exception {
     Webhook webhook = new Webhook(PROJECT_UUID, CE_TASK_UUID, "my-webhook", server.url("/ping").toString());
-    WebhookPayload payload = new WebhookPayload("P1", "{the payload}");
 
     server.shutdown();
-    WebhookDelivery delivery = newSender().call(webhook, payload);
+    WebhookDelivery delivery = newSender().call(webhook, PAYLOAD);
 
     assertThat(delivery.getHttpStatus()).isEmpty();
     assertThat(delivery.getDurationInMs()).isEmpty();
     // message can be "Failed to connect" or "connect timed out"
     assertThat(delivery.getErrorMessage().get()).contains("connect");
     assertThat(delivery.getAt()).isEqualTo(NOW);
     assertThat(delivery.getWebhook()).isSameAs(webhook);
-    assertThat(delivery.getPayload()).isSameAs(payload);
+    assertThat(delivery.getPayload()).isSameAs(PAYLOAD);
   }
 
   @Test
   public void silently_catch_error_when_url_is_incorrect() throws Exception {
     Webhook webhook = new Webhook(PROJECT_UUID, CE_TASK_UUID, "my-webhook", "this_is_not_an_url");
-    WebhookPayload payload = new WebhookPayload("P1", "{the payload}");
 
-    WebhookDelivery delivery = newSender().call(webhook, payload);
+    WebhookDelivery delivery = newSender().call(webhook, PAYLOAD);
 
     assertThat(delivery.getHttpStatus()).isEmpty();
     assertThat(delivery.getDurationInMs()).isEmpty();
     assertThat(delivery.getError().get()).isInstanceOf(IllegalArgumentException.class);
     assertThat(delivery.getErrorMessage().get()).isEqualTo("unexpected url: this_is_not_an_url");
     assertThat(delivery.getAt()).isEqualTo(NOW);
     assertThat(delivery.getWebhook()).isSameAs(webhook);
-    assertThat(delivery.getPayload()).isSameAs(payload);
+    assertThat(delivery.getPayload()).isSameAs(PAYLOAD);
+  }
+
+  /**
+   * SONAR-8799
+   */
+  @Test
+  public void redirects_should_be_followed_with_POST_method() throws Exception {
+    Webhook webhook = new Webhook(PROJECT_UUID, CE_TASK_UUID, "my-webhook", server.url("/redirect").toString());
+
+    // /redirect redirects to /target
+    server.enqueue(new MockResponse().setResponseCode(307).setHeader("Location", server.url("target")));
```

Comment:
I prefer not reusing the same constants as tested code because it highlights when a constant is badly used. Here the HTTP codes are quite common. Everybody knows them, don't you ? :-)